### PR TITLE
Upgrade aws library

### PR DESCRIPTION
**Pull request by MancunianSam**
The main changes are that we now need a custom http handler to work with our Cloudfront distribution. We use this to remove the bucket name from the path and to add xhrWithCredentials.
The tests have also changed. We can no longer directly influence the onHttpProgress function in the tests so some of the tests checking the upload progress have had to change slightly.

### CODE REVIEW

File: npm/src/s3upload/tdr-fetch-handler.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+  })
+}
+
+export interface FetchHttpHandlerOptions {
+  /**
+   * The number of milliseconds a request can take before being automatically
+   * terminated.
+   */
+  requestTimeout?: number
```
Comment (TomJKing) [inline npm/src/s3upload/tdr-fetch-handler.ts:23]: Maybe change the name of the param to this, just to make it a bit clearer is milliseconds: `requestTimeoutMs`

File: npm/src/s3upload/tdr-fetch-handler.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const { port, method } = request
+    const url = `${request.protocol}//${request.hostname}${
+      port ? `:${port}` : ""
+    }${path}`
+    // Request constructor doesn't allow GET/HEAD request with body
+    // ref: https://github.com/whatwg/fetch/issues/551
+    const body =
+      method === "GET" || method === "HEAD" ? undefined : request.body
+
+    const requestOptions: RequestInit = {//eslint-disable-line
```
Comment (TomJKing) [inline npm/src/s3upload/tdr-fetch-handler.ts:68]: Is this comment `//eslint-disable-line` needed?

File: npm/src/upload/index.ts [hunk 3]
```diff
@@ -37,9 +40,21 @@ export class FileUploader {
     goToNextPage: () => void,
     keycloak: KeycloakInstance
   ) {
+    const requestTimeout = 20 * 60 * 1000
```
Comment (TomJKing) [inline npm/src/upload/index.ts:43]: Maybe change param name to `requestTimeoutMs`?

**Review by TomJKing — COMMENTED** [review]
What can I say, nothing like making things straightforward! ;) Just a couple of minor comments. Feel free to ignore.

File: npm/src/s3upload/tdr-fetch-handler.ts (hunk 2 above)
Comment (MancunianSam) [reply npm/src/s3upload/tdr-fetch-handler.ts:68]: The RequestInit is part of the base typescript dom library so it doesn't need an explicit import which eslint doesn't like. I had a google though and you can add it to the .eslintrc and get rid of the comment so I've done that.

File: npm/src/s3upload/tdr-fetch-handler.ts (hunk 1 above)
Comment (MancunianSam) [reply npm/src/s3upload/tdr-fetch-handler.ts:23]: Yeah makes sense, I've changed it here and where it's called.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**